catchpoint_opentsdb_bridge.py

```diff
@@ -119,29 +142,21 @@ def post(self):
     ap.add_argument('-p', '--tsdb-port', default=4242, help='OpenTSDB port', type=int)
     ap.add_argument('-k', '--key', help='Catchpoint Push API key', required=True)
     ap.add_argument('-s' ,'--secret', help='Catchpoint Push API secret', required=True)
+    ap.add_argument('-f' ,'--file-mode', action='store_true', help='Dump Catchpoint data to the local filesystem and do not connect to OpenTSDB')
     args = ap.parse_args()
 
-    l = logging.getLogger()
-    l.setLevel(loglevel)
-    format_str = '[cp_pushd' \
-        + ':%(module)s:%(name)s:%(lineno)d] ' \
-        '%(levelname)s: %(message)s'
-    term_format = logging.Formatter(fmt='%(asctime)s ' + format_str)
-
-    # log to stdout for now
-    h = logging.StreamHandler(sys.stdout)
-    h.setFormatter(term_format)
-    l.addHandler(h)
-
-    obs = twisted_log.PythonLoggingObserver()
-    obs.start()
-
     # init catchpoint node list
     nodes = catchpoint_nodes(args.key, args.secret)
     nodes.update_node_list()
 
-    tsdb = OpenTSDBFactory()
-    reactor.connectTCP(args.tsdb_host, args.tsdb_port, tsdb)
+    if args.file_mode:
+        l.info('File mode detected. Not connecting to OpenTSDB')
+        l.info("Printing Catchpoint data to local log file {log_file}".format(log_file=log_file))
+        tsdb = None
+    else:
+        l.info('Connecting to OpenTSDB')
+        tsdb = OpenTSDBFactory()
+        reactor.connectTCP(args.tsdb_host, args.tsdb_port, tsdb)
```

Comment:
Not sure on the requirements here, but any chance we would want to TEE the data between both tsdb and a file?

Reply:
The requirements that I have are to write to a logfile for testing and OpenTSDB maintenance windows (or in the event of replacement of that data store). I can see the need to write to both places, though I don't have a use case for that, but I can add that functionality if need be.

Also, I need to update the README to describe the end state of whatever gets implemented.

Thoughts?

Reply:
ah ok no worries at all, I was just curious what the actual end goal of it was, if there is no need to do the TEE then its probably worth it.